tests/jupyterhub_test.py

```diff
@@ -15,6 +15,22 @@ def setup_module(module):
     )
 
 
+def test_api_security_without_role(fc):
+    from pykern.pkunit import pkok
+    from sirepo.pkcli import roles
+    import sirepo.auth_role
+
+    x = sirepo.auth_role.for_sim_type("jupyterhublogin")
+    roles.delete(fc.sr_uid, x)
+    pkok(x not in fc.sr_auth_state().roles, f"{x} role should have been removed")
```

Comment:
don't use f-strings. allow pkdebug to handle: `"{} role should...", x`

Reply:
[3a9f0d1](https://github.com/radiasoft/sirepo/pull/7027/commits/3a9f0d1399801354cea9e77a661893218636e2dd)